cypress/e2e/devpool.cy.ts

```diff
@@ -0,0 +1,146 @@
+    cy.get("#issues-container > :nth-child(1)").should("have.class", "new-task");
+
+    // needed to make sure data is written to the local storage
+    cy.wait(3000);
+
+    // Should display still one old task
+    cy.log("Should display still one old task");
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 200,
+        body: [issue1, issue2],
+      });
+    }).as("getIssues");
+    cy.visit("/");
+    cy.get('div[id="issues-container"]').children().should("have.length", 1);
+    cy.get("#issues-container > :nth-child(1)").should("not.have.class", "new-task");
+
+    // needed to make sure data is written to the local storage
+    cy.wait(3000);
+
+    cy.log("Should display two new tasks");
+    cy.clock(Date.now() + 95000000);
+    cy.visit("/");
+    const fakeNow = new Date("2022-04-10");
+    // Needed due to a bug
+    cy.clock(fakeNow).then((clock) => {
+      // @ts-expect-error https://github.com/cypress-io/cypress/issues/7577
+      return clock.bind(window);
+    });
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+  });
+
+  it("Display a message on rate limited", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 403,
+      });
+    }).as("getIssues");
+    cy.visit("/");
+    cy.get(".preview-header").should("exist");
+  });
+
+  it("Items can be sorted", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 200,
+        body: [issue1, issue2],
+      });
+    }).as("getIssues");
+    cy.visit("/");
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="price"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="price"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="time"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="time"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="priority"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="priority"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="activity"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="activity"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get("#filter").type("draft");
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+  });
+
+  it("User can log in", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
```

Comment:
If you are intercepting everything doesn't that kill the point of the test? Why don't we do real end-to-end tests? What if Supabase is broken and we can't oauth etc

Reply:
Since we don't develop the other end (eg the backend) I didn't think it was relevant to test it. If GitHub backend breaks there is nothing we can do. The proper way to handle that would be to have BE error test cases, and to have a health check running on the production instance.

Reply:
Okay sounds smart I'm in. You can file some issues for this with sufficient details so we can get around to it?

Reply:
You mean an issue mentioning the creation of e2e testing with the real endpoint correct?

Reply:
The proper way as you say